### PR TITLE
Added some additional comments to version service.

### PR DESCRIPTION
**Pull request by justinlittman**

### CODE REVIEW

File: app/services/version_service.rb [hunk 1]
```diff
@@ -20,10 +20,15 @@ def initialize(work)
   # @option opts [Hash] :vers_md_upd_info If present, used to add to the events datastream and set the desc and significance on the versionMetadata datastream
   # @raise [Dor::Exception] if the object hasn't been accessioned, or if a version is already opened
   def open(opts = {})
+    # Object has been accessioned at least once.
```
Comment (jcoyne) [inline app/services/version_service.rb]: I'm not sure what this means.  Does this mean "this method should only be called on an object that has been accessioned at least once?"

File: app/services/version_service.rb [hunk 2]
```diff
@@ -20,10 +20,15 @@ def initialize(work)
...
     # During local development, we need a way to open a new version even if the object has not been accessioned.
     raise(Dor::Exception, 'Object net yet accessioned') unless
       opts[:assume_accessioned] || Dor::Config.workflow.client.lifecycle('dor', work.pid, 'accessioned')
+    # Raised if for the current version, there are are any incomplete wf steps and there is a versionWF.
```
Comment (jcoyne) [inline app/services/version_service.rb]: Rephrase "Raised when the current version has any incomplete workflow steps and there is a versionWF"

File: app/services/version_service.rb [hunk 3]
```diff
@@ -20,10 +20,15 @@ def initialize(work)
...
+    # The open lifecyle is part of the versioningWF.
     raise Dor::VersionAlreadyOpenError, 'Object already opened for versioning' if open?
+    # Raised if for the current version, there are any incomplete wf steps and there is an accessionWF.
+    # The submittedWF is part of the accessionWF.
```
Comment (jcoyne) [inline app/services/version_service.rb]: `s/submittedWF/submitted milestone`

File: app/services/version_service.rb [hunk 4]
```diff
@@ -20,10 +20,15 @@ def initialize(work)
...
   # @raise [Dor::Exception] if the object hasn't been accessioned, or if a version is already opened
   def open(opts = {})
+    # Object has been accessioned at least once.
     # During local development, we need a way to open a new version even if the object has not been accessioned.
     raise(Dor::Exception, 'Object net yet accessioned') unless
       opts[:assume_accessioned] || Dor::Config.workflow.client.lifecycle('dor', work.pid, 'accessioned')
+    # Raised if for the current version, there are are any incomplete wf steps and there is a versionWF.
+    # The open lifecyle is part of the versioningWF.
     raise Dor::VersionAlreadyOpenError, 'Object already opened for versioning' if open?
+    # Raised if for the current version, there are any incomplete wf steps and there is an accessionWF.
```
Comment (jcoyne) [inline app/services/version_service.rb]: Can this be rephrased to "Raised when the current version has any incomplete workflow steps and there is an accessionWF"

**Comment by justinlittman** [comment]
@jcoyne Fixed all.

**Pull request merged**